src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,27 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch
+                {
+                    //TODO What is the correct behaviour for discarding exceptions
+                    // in this situation?  Should we log them in any way?
+                }
+            }
+
             // This is the blocking method that waits for an event to occur.  It wraps the SDK's WaitCommEvent function.
             [SuppressMessage("Microsoft.Interoperability", "CA1404:CallGetLastErrorImmediatelyAfterPInvoke", Justification = "this is debug-only code")]
```

Comment:
Certainly we suppress invalid code analyis violations. I can't say whether we typically run this rule or not but it's fine to keep it. At some point someone needs to decide whether to change the code or fix the message.